zapcore/encoder.go

```diff
@@ -80,16 +182,32 @@ type ObjectEncoder interface {
 // arrays even though they aren't typical in Go. Like slices, ArrayEncoders
 // aren't safe for concurrent use (though typical use shouldn't require locks).
 type ArrayEncoder interface {
+	// Built-in types
+	PrimitiveArrayEncoder
```

Comment:
I'll break out this change into a separate PR - it was an oversight on my part. Passing an `ArrayEncoder` to a `TimeEncoder` opens us up to infinite recursion :/